source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,41 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+void BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
+  val = buffer.peekBEInt<int32_t>(offset);
+  offset += sizeof(int32_t);
+}
+
+void BufferHelper::peekInt64(Buffer::Instance& buffer, uint64_t& offset, int64_t& val) {
+  val = buffer.peekBEInt<int64_t>(offset);
+  offset += sizeof(int64_t);
+}
+
+void BufferHelper::peekBool(Buffer::Instance& buffer, uint64_t& offset, bool& val) {
+  char byte = buffer.peekInt<char, ByteOrder::Host, 1>(offset);
+  val = static_cast<bool>(byte & 255);
+  offset += 1;
+}
+
+void BufferHelper::peekString(Buffer::Instance& buffer, uint64_t& offset, std::string& str) {
+  int32_t len = 0;
+  peekInt32(buffer, offset, len);
+
+  if (buffer.length() < (offset + len) || !len) {
+    return;
```

Comment:
Should we be throwing here? Or is it reasonable to expect parsing to still succeed if we hit this?

Reply:
Hmm, for `len == 0` we shoud silently return since an empty string is valid. For `buffer.length() < offset + len`, we shoud probably throw and forget it.